### DISCUSSION TITLE
Using nameof to get name of tuple field.

### DISCUSSION THREAD

**Discussion by MichaelStett** [dotnet/csharplang#3570] [General]
When trying to get `nameof` of tuple field directly through element from List i encountered such problem:
`Error CS8082	Sub-expression cannot be used in an argument to nameof.`

I am currently using dotnet in verision 5.0.100-preview.4.20258.7

```C#
using System;
using System.Collections.Generic;

namespace Issue
{
    class Program
    {
        static void Main(string[] args)
        {
            var items = new List<(int Id, string Name)>
            {
                 (0, "Test-0"),
                 (1, "Test-1"),
                 (2, "Test-2"),
            };

            var item = items[0];

            // Works fine
            Console.WriteLine(nameof(item.Name));

            // Error: Sub-expression cannot be used in an argument to nameof.
            Console.WriteLine(nameof(items[0].Name)); 
        }
    }
}
```

**Comment by CyrusNajmabadi** [comment]
This feels like a language design request.  Those would go to dotnet/csharplang

**Comment by myblindy** [comment]
Only if it was intentional that this doesn't work. Is it?

**Comment by gafter** [comment]
Nameof is only specified to accept a (possibly qualified) name, not an arbitrary expression.
